### DISCUSSION TITLE
Can't revoke token

### DISCUSSION THREAD

**Discussion by gandaldf** [zitadel/oidc#424] [Q&A]
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe the docs your are missing or that are wrong

I am developing a client that uses the Relaying Party to interface with the Zitadel portal, following the example: https://github.com/zitadel/oidc/blob/main/example/client/app/app.go

During the login phase, just after calling the UserinfoCallback, I save a session with my user's id_token obtained from tokens `*oidc.Tokens[*oidc.IDTokenClaims]`:
`
idToken := tokens.IDToken`

At the time of logout, I would expect, before deleting my session, to also revoke the associated tokens; therefore, I used this code:
```
if idToken := session.Get("id_token"); idToken != nil {
    rp.RevokeToken(myRP, idToken.(string), "id_token")
    rp.EndSession(myRP, idToken.(string), "http://localhost:3000", "")
}
```
But neither of the two functions seems to work.
If I check the events page in the Zitadel console, I can see the trace of the creation, but no trace of the token revocation. What am I doing wrong?
It would be useful to integrate the example above by also adding a handler for logout to clarify the issue for others who will approach the problem in the future...

**Comment by muhlemmer** [comment]
You should provide a refresh or access token:

https://zitadel.com/docs/apis/openidoauth/endpoints#revocation_endpoint

**Comment by gandaldf** [comment]
Hello @muhlemmer, I have created an OIDC/PKCE application, and the Zitadel panel does not allow me to save the response type as "ID Token" only "Code"; the only additional thing I can select is "Refresh token"...
However, when I observe the tokens in response from the UserinfoCallback, I only see the ID Token... no Refresh token.
I tried to revoke my session using the following code:
rp.RevokeToken(a.Provider, idToken.(string), "id_token")
rp.EndSession(a.Provider, idToken.(string), a.config.LogoutURL, "")
but unfortunately, it doesn't work.

**Reply by muhlemmer** [reply]
Sorry for the slow replies, I'm currently traveling.

It would help a lot if you share the code that  you are using to obtain the and revoke tokens as a [Minimal Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

A public repository with the code would also be ok. (Please permalink the lines).
